subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/ImpactMetrics.kt

```diff
@@ -0,0 +1,22 @@
+package com.avito.ci.steps
+
+import com.avito.kotlin.dsl.isRoot
+import com.avito.kotlin.dsl.namedOrNull
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskProvider
+
+class ImpactMetrics(context: String, name: String) : BuildStep(context, name) {
+
+    override fun registerTask(project: Project, rootTask: TaskProvider<out Task>) {
+        require(project.isRoot()) { failMessage("should be configured for root project") }
```

Comment:
How could we add this step to the root project?

Reply:
I don't see restrictions for the root project. It will add a task to it, e.g. `:fastCheck`